python/lsst/daf/butler/datastores/fileDatastore.py

```diff
@@ -2609,42 +2626,47 @@ def emptyTrash(self, ignore_errors: bool = True) -> None:
                 # Only trashed refs still known to datastore will be returned.
                 location = info.file_location(self.locationFactory)
 
-                # Point of no return for this artifact
-                log.debug("Removing artifact %s from datastore %s", location.uri, self.name)
-                try:
-                    self._delete_artifact(location)
-                except FileNotFoundError:
-                    # If the file itself has been deleted there is nothing
-                    # we can do about it. It is possible that trash has
-                    # been run in parallel in another process or someone
-                    # decided to delete the file. It is unlikely to come
-                    # back and so we should still continue with the removal
-                    # of the entry from the trash table. It is also possible
-                    # we removed it in a previous iteration if it was
-                    # a multi-dataset artifact. The delete artifact method
-                    # will log a debug message in this scenario.
-                    # Distinguishing file missing before trash started and
-                    # file already removed previously as part of this trash
-                    # is not worth the distinction with regards to potential
-                    # memory cost.
-                    pass
-                except Exception as e:
-                    if ignore_errors:
-                        # Use a debug message here even though it's not
-                        # a good situation. In some cases this can be
-                        # caused by a race between user A and user B
-                        # and neither of them has permissions for the
-                        # other's files. Butler does not know about users
-                        # and trash has no idea what collections these
-                        # files were in (without guessing from a path).
-                        log.debug(
-                            "Encountered error removing artifact %s from datastore %s: %s",
-                            location.uri,
-                            self.name,
-                            e,
-                        )
-                    else:
-                        raise
+                artifacts_to_delete.append(location)
+
+            # Now do the deleting.
+            log.debug("Now removing %d file artifacts from datastore %s", len(artifacts_to_delete), self.name)
+
+            try:
+                self._delete_artifacts(artifacts_to_delete)
+            except* FileNotFoundError:
```

Comment:
But this is handled already in `_delete_artifacts`?

Reply:
Yes. But this was a nice comment I didn't want to delete and I didn't think it would do any harm. Maybe I should move the `_delete_artifact` code into `emptyTrash` instead...